pkg/upgrade/util.go

```diff
@@ -0,0 +1,40 @@
+package upgrade
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	ofapi "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func GetOperatorNamespace() (string, error) {
+	data, err := os.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace")
+	if err == nil {
+		if ns := strings.TrimSpace(string(data)); len(ns) > 0 {
+			return ns, nil
+		}
+	}
+
+	return "", err
+}
```

Comment:
i think the reason for the original code is to convert []byte to string, so it wont need to be converted in the caller.
but i agree to check the err is not that necessary.
how about doing like this:
```
func GetOperatorNamespace() (string) {
	data, _ := os.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace")
	return string(data)
}
```

Reply:
I don't think swallowing an error is a good idea for most of the cases. I wasn't really suggesting it :)

Here we should return it as it might fail for other reasons - e.g. no read access for the process (though suspicious). If we want `string` to be returned how about simply doing this instead:

```
func GetOperatorNamespace() (string, error) {
	data, err := os.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace")
	return string(data), err
}
```